[WIP] tidb-cloud: update for Starter mode (#22887)
pingcap/docs: sql-statements/sql-statement-import-into.md

- For TiDB Self-Managed, each `IMPORT INTO` task supports importing data within 10 TiB. If you enable the [Global Sort](/tidb-global-sort.md) feature, each `IMPORT INTO` task supports importing data within 40 TiB.
- For [TiDB Cloud Dedicated](https://docs.pingcap.com/tidbcloud/select-cluster-tier#tidb-cloud-dedicated), if your data to be imported exceeds 500 GiB, it is recommended to use TiDB nodes with at least 16 cores and enable the [Global Sort](/tidb-global-sort.md) feature, then each `IMPORT INTO` task supports importing data within 40 TiB. If your data to be imported is within 500 GiB or if the cores of your TiDB nodes are less than 16, it is not recommended to enable the [Global Sort](/tidb-global-sort.md) feature.
- The execution of `IMPORT INTO ... FROM FILE` blocks the current connection until the import is completed. To execute the statement asynchronously, you can add the `DETACHED` option.

> - On [{{{ .starter }}}](https://docs.pingcap.com/tidbcloud/select-cluster-tier#starter) instances, TiDB prepares source files and performs pre-checks synchronously before creating an import job. This behavior also applies when `DETACHED` is specified. The `DETACHED` option only makes the import job run asynchronously after it is created. If the source data exceeds the [`starter-params.max-import-data-size`](/tidb-configuration-file.md#max-import-data-size) limit, the statement returns an error and no import job is created.

Reviewer: ```suggestion
- On [{{{ .starter }}}](https://docs.pingcap.com/tidbcloud/select-cluster-tier#starter) instances, the source data size cannot exceeds the [`starter-params.max-import-data-size`](/tidb-configuration-file.md#max-import-data-size) limit.
```

Author: Thanks for the suggestion! We kept a more detailed description here because, in Starter mode, TiDB prepares source files and runs pre-checks synchronously before creating the import job, even when `DETACHED` is specified. The `DETACHED` option only makes the job run asynchronously after creation, and an over-limit source is rejected before any job is created. The linked `starter-params.max-import-data-size` reference also documents that explicitly setting `0B` disables the limit.
